src/transformers/data/data_collator.py

```diff
@@ -133,3 +133,111 @@ def mask_tokens(self, inputs: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]
 
         # The rest of the time (10% of the time) we keep the masked input tokens unchanged
         return inputs, labels
+
+
+@dataclass
+class DataCollatorForXLNetLanguageModeling:
```

Comment:
Not 100p sure what a good name here is - maybe `PermutedAutoregressiveLanuageModeling` or we stick to `GeneralizedAutoregressiveLanguageModeling` or `PermutationSampledAutoregressiveLM` ...? What do you think is a good name? Would prefer to not make it model specific even though only XLNet does this kind of training at the moment. @thomwolf @TevenLeScao - do you have a good idea for a name?

Reply:
The authors have named the objective `PermutationLanguageModeling` (section 2.2 in the paper) - shall we stick with that?